### ISSUE TITLE
No success implementing related products component

### ISSUE THREAD

**Issue by haralake** [OFFLINE-GmbH/oc-mall-plugin#491]
I try to implement related products example and display four random items from the same category of the currently viewed product. I try this example given on:
https://offline-gmbh.github.io/oc-mall-plugin/components/products.html#example-implementations

```
title = "Product"
url = "/product/:slug/:variant?"
layout = "default"
is_hidden = 0

[product]
product = ":slug"
variant = ":slug"

[products relatedProducts]
setPageTitle = 0
includeVariants = 1
includeChildren = 0
perPage = 4
paginate = 0
sort = "random"
==
use OFFLINE\Mall\Models\Category;
function onStart() {
    // Fetch the category from the product component.
    $category = Category::find(optional($this->page->components['product']->item)->category_id);
    if ($category) {
        // If a category is available, use it for the products component.
        $this->page->components['products']->category = $category;
    }
}
==
{% component 'product' %}

<h2>Other products from this category</h2>
{% component 'relatedProducts' %}
```

but the products that displayed are from different categories. Any thoughts what is wrong?

**Comment by tobias-kuendig** [comment]
Try to see what the value of `$category` is: `dd($category);`

Also check if `$this->page->components['product']->item` returns something.

**Comment by haralake** [comment]
`$category = Category::find(optional($this->page->components['product']->item)->category_id);`
returns **Null** using `dd($category);`

So i used:

```
    $product    = $this['product']->product;
    $category   = $product->categories->first();

    if ($category) {
        // If a category is available, use it for the products component.
        $this->page->components['relatedProducts']->category = $category;
    }
```

and it seems to work well.

**Comment by tobias-kuendig** [comment]
Did you place the `product` component on a partial instead of a cms page?

The example code tries to get the component off the page so this would explain the error.

**Comment by haralake** [comment]
I'm using mall demo theme, product component is in cms page. Using:
`$category = Category::find(optional($this->page->components['product']->item)->category_id);`
to fetch the category from the product component seems that give Null category id.
I used
```
    $product    = $this['product']->product;
    $category   = $product->categories->first();
```
and is working as expected.

**Comment by az-vm** [comment]
Hello @haralake did you got something on this ?

**Comment by tobias-kuendig** [comment]
The example has been fixed in 120c9f5, thanks for the report!

**Comment by zlobec** [comment]
> Did you place the `product` component on a partial instead of a cms page?
> 
> The example code tries to get the component off the page so this would explain the error.

I'm trying the same objective from the StaticPages but the component 'products' is loaded from the partial, and it does load products, but the ajax functions do not work... ex. addToCart.

Any workaround?
